pkg/space/space_test.go

```diff
@@ -53,3 +53,30 @@ func TestNewSubSpace(t *testing.T) {
 	)
 	assert.Equal(t, expectedSubSpace, subSpace)
 }
+
+func TestNewSubSubSpace(t *testing.T) {
+	// given
+	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+	sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
+		spacerequesttest.WithTierName("appstudio"),
+		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+	parentSpace := spacetest.NewSpace(test.HostOperatorNs, "parentSpace")
+	subSpace := NewSubSpace(sr, parentSpace)
+
+	// when
+	subSubSpace := NewSubSpace(sr, subSpace)
```

Comment:
Let's not use the same SpaceRequest for both SubSpaces to make sure that the subSubSpace doesn't have the same name as its parent (subSpace)

Reply:
Fixed in 35eb34a.